ReactWindows/ReactNative/Modules/Camera/CameraRollManager.cs

```diff
@@ -0,0 +1,156 @@
+        /// </summary>
+        [ReactMethod]
+        public async void saveToCameraRoll(JObject options, IPromise promise)
+        {
+            // TODO: tag, type?
+            promise.Reject(new NotImplementedException("CameraRoll.saveToCameraRoll() method not yet implemented."));
+        }
+
+        /// <summary>
+        /// Returns a Promise with photo/video identifier objects from the local camera roll.
+        /// </summary>
+        [ReactMethod]
+        public async void getPhotos(JObject options, IPromise promise)
+        {                
+            try
+            {                 
+                if (options["first"] == null)
+                {
+                    promise.Reject(new ArgumentNullException("The option [first] must be defined."));
+                    return;
+                }
+
+                if ((int) options["first"] < 1)
+                {
+                    promise.Reject(new ArgumentException("The option [first] is not valid."));
+                    return;
+                }
+
+                var limit = (uint)(options["first"]);
+
+                // TODO:
+                // after
+                // mimeTypes, groupName 
+
+                if (options["groupTypes"] != null) {
+                    promise.Reject(new ArgumentException("The option [groupTypes] is not supported on UWP"));
+                    return;
+                }
+
+                QueryOptions queryOption;
+
+                switch ((string)(options["assetType"]))
+                {
+                    case "All":
+                        queryOption = new QueryOptions(CommonFileQuery.OrderByDate, SupportedVideoFileTypes.Concat(SupportedImageFileTypes).ToList());
+                        break;
+                    case "Videos":
+                        queryOption = new QueryOptions(CommonFileQuery.OrderByDate, SupportedVideoFileTypes);
+                        break;
+                    case "Photos":
+                    default:
+                        queryOption = new QueryOptions(CommonFileQuery.OrderByDate, SupportedImageFileTypes);
+                        break;
+                }
+
+                StorageFolder picturesFolder = KnownFolders.CameraRoll;
+                queryOption.FolderDepth = FolderDepth.Deep;
+
+                IReadOnlyList<StorageFile> sortedItems = await picturesFolder.CreateFileQueryWithOptions(queryOption).GetFilesAsync(0, limit);
+                
+                var result = new JObject();
+                var edges = new JArray();
+                var pageinfo = new JObject();
+                var endcursor = "";
+
+                foreach (StorageFile file in sortedItems)
+                {
+                    var node = new JObject();
+                    var nodeObj = new JObject();
+                    var image = new JObject();
+                    var location = new JObject();
+                    var uri = new Uri(file.Path);
```

Comment:
No need to allocate all these upfront. Some of them may not even be included in the result (e.g., `location`). Just as easy to allocate when needed (see suggestion below for location).